### ISSUE TITLE
suggest adding a mru-sorting/mfu-sorting algorithm

### ISSUE THREAD

**Issue by JessonChan** [fardavide/Shuttle#313]
I really love this mobile app, it provides very practical features. However, I suggest adding a sorting algorithm, which could be based on the frequency of most-used (mfu)or recently used (mru)items. Looking forward to your reply, thank you very much.

**Comment by fardavide** [comment]
Thank you for your feedback @JessonChan , it's very appreciated!

Currently the app already sorts by MFU, considering current location and time:
1) first all MFU for current location + time
2) then MFU for location only
3) MFU for time only 
4) lastly MFU with no constraints 

**About your mention to MFU**
Are you suggesting another option to sort only by MFU without time and location constraints?

**About your mention to MRU **
The current database structure doesn't support that; here's the rationale:
With the previous implementation, we were storing an entry every time an app is opened, including time and location; this approach was very versatile, but had a very big performance penalty: since we would have a ridiculous amount of rows in the database, the sorting algorithm would take up to a couple of seconds to run. To solve this, we could cache a sorted list, but it would kill a big feature of the app, which is real-time sorting: a real life problem that I faced day to day, was having instantly updated suggestions, based on my location: for example, as a user, I expect to have my fidelity card app at the first place as soon as I reach the shopping mall.
For this reason, we dropped that idea and chose instead to create "pre-packed" containers, related to various combinations of time and location, formed by the id of a given up and a counter to sort by.
This allowed us a big boost in terms of performance, but a stricter range of flexibility.

The feature that you're requesting it's out of the scope of the app -since other solutions have simple sorting by MRU- and would require a separate database table, impacting on performance and maintainability of the code. 
I'm not saying we don't want to implement that, but it would fall into a low priority task, especially considering that I'm the only developer working on this in my free time; but any contribution would be **very** appreciated!
